cli/src/lib.rs

```diff
@@ -1954,6 +1969,31 @@ fn validator_flags(cfg: &WithPath<Config>) -> Result<Vec<String>> {
                         // Push the clone flag for each array entry
                         flags.push("--clone".to_string());
                         flags.push(entry["address"].as_str().unwrap().to_string());
+
+                        // Check it is a program account
+                        let pubkey = Pubkey::from_str(flags.last().unwrap())
+                            .map_err(|_| anyhow!("Invalid pubkey {}", flags.last().unwrap()))?;
+
+                        if let Some(ref client) = client {
+                            let account = client
+                                .get_account_with_commitment(&pubkey, CommitmentConfig::default())?
+                                .value
+                                .map_or(Err(anyhow!("Account {} not found", pubkey)), Ok)?;
+
+                            if account.owner == bpf_loader_upgradeable::id() {
+                                let upgradable: UpgradeableLoaderState = account
+                                    .deserialize_data()
+                                    .map_err(|_| anyhow!("Invalid program account {}", pubkey))?;
+
+                                if let UpgradeableLoaderState::Program {
+                                    programdata_address,
+                                } = upgradable
+                                {
+                                    flags.push("--clone".to_string());
+                                    flags.push(programdata_address.to_string());
```

Comment:
this should only be pushed if it's not in `value.as_array()` anyway

Reply:
noted, i have added this condition too